client/dashboard/sites/site-reset-modal/index.tsx

```diff
@@ -153,14 +153,12 @@ export default function SiteResetModal( { site, onClose }: { site: Site; onClose
 	const { createSuccessNotice } = useDispatch( noticesStore );
 	const [ error, setError ] = useState< string | null >( null );
 
-	const { data: siteContentSummary } = useQuery< SiteResetContentSummary >(
-		siteResetContentSummaryQuery( site.ID )
-	);
+	const { data: siteContentSummary } = useQuery( siteResetContentSummaryQuery( site.ID ) );
 
 	const statusQuery = {
 		...siteResetStatusQuery( site.ID ),
 		...( site.is_wpcom_atomic && {
-			refetchInterval: ( query: Query< SiteResetStatus > ) => {
+			refetchInterval: ( query: { state: { data?: SiteResetStatus } } ) => {
```

Comment:
This is a great example of how the "extract query options" pattern allows powerful composition.

However this composition is so fancy that we need to add types unfortunately 😢 Rather than use the parameters real type, I'm wondering if we will have more luck if we make the type as loose as possible: any object that just happens to have a `state` object in it.